src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -608,20 +756,62 @@ protected DenseMatrix initContingencyRhs(LfNetwork lfNetwork, EquationSystem equ
+            Set<ComputedContingencyElement> groupOfElementsBreakingConnectivity = getGroupOfElementsBreakingConnectivity(lfNetwork, states,
                     contingency.getElements().stream().map(element -> contingenciesElements.get(element.getId())).collect(Collectors.toList()), equationSystem);
             if (groupOfElementsBreakingConnectivity.isEmpty()) { // connectivity not broken
                 nonLosingConnectivityContingencies.add(contingency);
-            } else if (groupOfElementsBreakingConnectivity.stream().mapToInt(Set::size).max().getAsInt() > 1) {
-                throw new PowsyblException("The contingency " + contingency.getId() + " breaks the connectivity on more than one branch.");
             } else {
                 contingenciesByGroupOfElementsBreakingConnectivity.computeIfAbsent(groupOfElementsBreakingConnectivity, key -> new LinkedList<>()).add(contingency);
             }
         }
     }
 
+    private Set<String> getElementsToReconnect(GraphDecrementalConnectivity<LfBus> connectivity, Set<ComputedContingencyElement> breakingConnectivityCandidates) {
+        Set<String> elementsToReconnect = new HashSet<>();
+
+        Map<Pair<Integer, Integer>, Set<ComputedContingencyElement>> elementsByConnectedComponents =
+                breakingConnectivityCandidates.stream().collect(Collectors.toMap(
+                    element -> Pair.of(connectivity.getComponentNumber(element.getLfBranch().getBus1()), connectivity.getComponentNumber(element.getLfBranch().getBus2())),
+                    element -> new HashSet<>(Collections.singletonList(element)),
+                    (existing, replacement) -> {
+                        existing.addAll(replacement);
+                        return existing;
+                    }
+                ));
+
+        Map<Integer, Set<Integer>> connections = new HashMap<>();
+        connections.put(0, Collections.singleton(0));
+        for (Set<LfBus> smallComponent : connectivity.getSmallComponents()) {
+            int ccNumber = connectivity.getComponentNumber(smallComponent.iterator().next());
+            connections.put(ccNumber, new HashSet<>(Collections.singletonList(ccNumber)));
+        }
+
+        for (Map.Entry<Pair<Integer, Integer>, Set<ComputedContingencyElement>> elementsByCc : elementsByConnectedComponents.entrySet()) {
+            Integer cc1 = elementsByCc.getKey().getKey();
+            Integer cc2 = elementsByCc.getKey().getValue();
+            if (connections.get(cc1).contains(cc2)) {
+                // cc are already connected
+                continue;
+            }
+            elementsToReconnect.add(elementsByCc.getValue().iterator().next().getElement().getId());
+            Set<Integer> newCc = new HashSet<>();
+            newCc.addAll(connections.get(cc1));
+            newCc.addAll(connections.get(cc2));
+            connections.get(cc1).forEach(integer -> connections.put(integer, newCc));
```

Comment:
Why is it only for integers in `connections.get(cc1)`? Shouldn't it be `newCc.forEach()` instead?